Zig.gitignore

```diff
@@ -0,0 +1,4 @@
+# This file is for zig-specific build artifacts.
```

Comment:
Do we really need this comment?

Reply:
I don't think so. All directories included in this `.gitignore` even have a `zig-` prefix.
I would just make it
```
zig-cache/
zig-out/
```
without anything else.

Reply:
Does it really matter? Probably not. 
Do I care to setup my machine to make the change? Nope. 
Why not? Because GitHub obviously doesn't give a f**k :) 

Maybe go vote for this PR #4294 in case he gets lucky.

I'm just a bitter old man by now.

Reply:
That PR is missing a trailing newline.

But I mean who uses these templates anyway? I definitely won't go here to copy two lines for a file I could write in 10 seconds. I'm guessing the main reason people want to see it merged is for publicity for Zig.

Reply:
@r00ster91 
When creating a project on GitHub, you can choose the programming language used for the project, and this is automatically generated, similar to your LICENSE file. `.gitignore` is used for that. :)

Reply:
> That PR is missing a trailing newline.
> 
> But I mean who uses these templates anyway? I definitely won't go here to copy two lines for a file I could write in 10 seconds. I'm guessing the main reason people want to see it merged is for publicity for Zig.

I'm surely not here to copy gitignore files, but as @Hanaasagi mentions, if it's in a dropdown I'd rather just select it there than writing the three-line (including newline) file! 😅

Reply:
As someone who is entirely new to Zig (like a lot of people), one of the first questions I ask is "what do I need to ignore?"

It is definitely nice to have that documented somewhere and I usually do pull it off GitHub if the built-in tooling/project template does not generate one.